okhttp/src/test/java/okhttp3/PublicInternalApiTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3;
+
+import okhttp3.internal.http.HttpHeaders;
+import okhttp3.internal.http.HttpMethod;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@SuppressWarnings("ALL") public class PublicInternalApiTest {
+  @Test public void permitsRequestBody() {
+    assertTrue(HttpMethod.permitsRequestBody("POST"));
+    assertFalse(HttpMethod.permitsRequestBody("GET"));
+  }
+
+  @Test public void requiresRequestBody() {
+    assertTrue(HttpMethod.requiresRequestBody("PUT"));
```

Comment:
nit: Given that `requiresRequestBody` is a public method and there aren't many test cases, it'd be okay to test all the assertTrue cases. i.e.
```
    assertTrue(HttpMethod.requiresRequestBody("POST"));
    assertTrue(HttpMethod.requiresRequestBody("PUT"));
    assertTrue(HttpMethod.requiresRequestBody("PATCH"));
    assertTrue(HttpMethod.requiresRequestBody("PROPPATCH"));
    assertTrue(HttpMethod.requiresRequestBody("REPORT"));
```

Reply:
I deliberately didn't do that because the test is a pass through and so I'd be copying other tests verbatim, seems like maintenance for not much gain.